node/config.test.ts

```diff
@@ -231,10 +232,63 @@ test('Loads function paths from the in-source `config` function', async () => {
   expect(postCacheRoutes.length).toBe(1)
   expect(postCacheRoutes[0]).toEqual({ function: 'user-func4', pattern: '^/user-func4/?$' })
 
-  expect(Object.keys(functionConfig)).toHaveLength(1)
+  expect(Object.keys(functionConfig)).toHaveLength(2)
   expect(functionConfig['user-func5']).toEqual({
     excluded_patterns: ['^/user-func5/excluded/?$'],
   })
+  expect(functionConfig['user-func2']).toEqual({
+    excluded_patterns: ['^/user-func2/excluded/?$'],
+  })
+
+  await cleanup()
+})
+
+test('Stores excludedPath in `routes` (flagged)', async () => {
+  const { basePath, cleanup, distPath } = await useFixture('with_config')
+  const userDirectory = resolve(basePath, 'netlify', 'edge-functions')
+  const internalDirectory = resolve(basePath, '.netlify', 'edge-functions')
+  const declarations: Declaration[] = [
+    {
+      function: 'framework-func2',
+      path: '/framework-func2',
+    },
+    {
+      function: 'user-func2',
+      path: '/user-func2',
+      excludedPath: '/user-func2/excluded',
+    },
+  ]
+  await bundle([internalDirectory, userDirectory], distPath, declarations, {
+    basePath,
+    configPath: join(internalDirectory, 'config.json'),
+    featureFlags: { edge_functions_excluded_patterns_on_route: true },
+  })
+
+  const manifestFile = await fs.readFile(resolve(distPath, 'manifest.json'), 'utf8')
+  const { routes, post_cache_routes: postCacheRoutes, function_config: functionConfig } = JSON.parse(manifestFile)
+
+  expect(routes).toEqual([
+    { function: 'framework-func2', pattern: '^/framework-func2/?$' },
+    {
+      function: 'user-func2',
+      pattern: '^/user-func2/?$',
+      // excludedPath is defined in TOML, so it's stored in `routes`
+      excluded_patterns: ['^/user-func2/excluded/?$'],
+    },
+    { function: 'framework-func1', pattern: '^/framework-func1/?$' },
+    { function: 'user-func1', pattern: '^/user-func1/?$' },
+    { function: 'user-func3', pattern: '^/user-func3/?$' },
+    // excludedPath is defined in ISC, so it's not here but in in `function_config`
```

Comment:
If what we find in ISC generates one entry in `routes`, could the excluded path also make its way there?

Reply:
We could do that for clarity, but it'd mean that data is present in both places (we still need it in `function_config` since it applies to all functions). I'd leave it out 🤷‍♂️ 

I think what's a bit weird here is that in the ISC `{ path: "/*", excludedPath: ["/*.css", "/*.js"] }`, `excludedPath` not only applies to `/*`, but also to _all_ other routes defined in the TOML, whereas an `excludedPath` defined in the TOML only applies to its accompanying `path`. That makes the mental model not as straight-forward to explain, but I think it can be expressed in docs something like this:



```md
# Edge Functions Routing

To assign an Edge Function to a specific route,
set the `path` field in your function config to the URL pattern
you want to apply the function to.
You can exclude certain paths, like static assets, using `excludedPath`.
This means that the function will never be run for those assets.

You can use `netlify.toml` to define more routes.
Again, set the `path` field to the URL pattern you want to apply the function to.
If you want to skip subpath, use `excludedPath`.
The excluded paths won't apply to the whole function, but only to that path.
This can be used to skip an edge function for certain subpages.
```

Reply:
It's interesting that you say the exclusions in ISC apply to every route of that function. I could also see us saying that ISC exclusions apply to the paths defined in ISC, and so the exclusions would go to all the routes generated from ISC. If that makes sense?